File: javascript/maximumdepthofbinarytree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    function depth(node){
        if (node===null){
            return 0
        }
        else{
            var leftdepth=depth(node.left);
            var rightdepth=depth(node.right);
        }
        return Math.max(leftdepth,rightdepth)+1;
    }
    return depth(root)   
};

File: python/maximumdepthofbinarytree.py
# https://leetcode.com/problems/maximum-depth-of-binary-tree/
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def depth(node):
            if node is None:
                return 0
            else:
                leftdepth=depth(node.left)
                rightdepth=depth(node.right)

            return max(leftdepth,rightdepth)+1
        return depth(root)
        